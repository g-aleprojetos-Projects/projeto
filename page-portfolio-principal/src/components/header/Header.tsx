import React, {useState} from 'react';
import * as S from './Header.styles';
import {Logo} from 'components/logo';
import {NavBar} from 'components/NavBar';

export const Header = () => {
  const [menuLateral, setMenuLateral] = useState<boolean>(false);

  const handleMudarMenu = () => {
    setMenuLateral(!menuLateral);
  };
  return (
    <S.ContainerHeader data-testid={'test_header'}>
      <S.ContainerPrincipal>
        <S.ContainerLogo>
          <Logo />
        </S.ContainerLogo>
        <S.ContainerNav>
          <NavBar openRightNav={menuLateral} mudarEstado={handleMudarMenu} />
        </S.ContainerNav>
      </S.ContainerPrincipal>
    </S.ContainerHeader>
  );
};
