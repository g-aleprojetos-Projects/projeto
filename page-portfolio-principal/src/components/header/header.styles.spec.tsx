import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './Header.styles';

describe('Header.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerHeader />).toJSON();
    expect(container).toMatchSnapshot();
  });

  test('Overlay DEVE ser igual ao snapshot', () => {
    const overlay = renderer.create(<S.ContainerPrincipal />).toJSON();
    expect(overlay).toMatchSnapshot();
  });

  test('Content DEVE ser igual ao snapshot', () => {
    const content = renderer.create(<S.ContainerContent />).toJSON();
    expect(content).toMatchSnapshot();
  });
});
