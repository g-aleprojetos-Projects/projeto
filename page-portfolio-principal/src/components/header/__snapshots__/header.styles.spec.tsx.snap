// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 9rem;
  background-color: 'transparent';
}

<div
  className="c0"
/>
`;

exports[`Header.styles Content DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 2.125rem;
  top: 3.438rem;
}

<div
  className="c0"
/>
`;

exports[`Header.styles Overlay DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 9rem;
  background-color: #00000080;
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;

exports[`Header.styles Video DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100vw;
  height: 100%;
  object-fit: cover;
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

<video
  className="c0"
  src="video.mp4"
/>
`;
